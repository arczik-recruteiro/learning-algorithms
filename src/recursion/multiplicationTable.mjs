'use strict';

export const multiplicationTable = (arr = []) => {
  if (arr.length === 0) {
    return;
  }

  let i = 0;
  const len = arr.length;
  const result = [];

  do {
    let j = 0;
    result.push(`==== ${arr.at(i)} ====`);
    do {
      const paragraph = `${arr.at(i)} * ${arr.at(j)}=${arr.at(i) * arr.at(j)}`;

      result.push(paragraph);
    } while (++j < len);
  } while (++i < len);

  return result;
};
