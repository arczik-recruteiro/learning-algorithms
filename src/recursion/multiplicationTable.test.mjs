import { multiplicationTable } from './multiplicationTable.mjs';

describe('multiplicationTable', () => {
  it('should print the result in console', () => {
    // console.log([])

    // console.log()

    const result1 = multiplicationTable([]);
    const result2 = multiplicationTable([1]);
    const result3 = multiplicationTable([2, 3, 5, 2, 5]);

    expect(result1).toBeUndefined();
    expect(result2).toMatchSnapshot();
    expect(result3).toMatchSnapshot();
  });
});
