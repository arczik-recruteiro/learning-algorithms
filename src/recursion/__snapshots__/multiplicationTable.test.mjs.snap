// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`multiplicationTable should print the result in console 1`] = `
[
  "1 * 1=1",
]
`;

exports[`multiplicationTable should print the result in console 2`] = `
[
  "2 * 2=4",
  "2 * 3=6",
  "2 * 5=10",
  "2 * 2=4",
  "2 * 5=10",
  "3 * undefined=NaN",
  "5 * undefined=NaN",
  "2 * undefined=NaN",
  "5 * undefined=NaN",
]
`;
